homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,222 @@
+                    data={},
+                    options={
+                        CONF_API_KEY: self.data[CONF_API_KEY],
+                        CONF_MAIN_USER: self.data[CONF_MAIN_USER],
+                        CONF_USERS: [
+                            self.data[CONF_MAIN_USER],
+                            *user_input[CONF_USERS],
+                        ],
+                    },
+                )
+        try:
+            main_user = get_lastfm_user(
+                self.data[CONF_API_KEY], self.data[CONF_MAIN_USER]
+            )
+            friends = [
+                SelectOptionDict(value=friend.name, label=friend.get_name(True))
+                for friend in main_user.get_friends()
+            ]
+        except WSError:
+            friends = []
+        return self.async_show_form(
+            step_id="friends",
+            errors=errors,
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_USERS): SelectSelector(
+                            SelectSelectorConfig(
+                                options=friends, custom_value=True, multiple=True
+                            )
+                        ),
+                    }
+                ),
+                user_input or {CONF_USERS: []},
+            ),
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Import config from yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_API_KEY] == import_config[CONF_API_KEY]:
+                return self.async_abort(reason="already_configured")
+        return self.async_create_entry(
+            title="LastFM",
+            data={},
+            options={
+                CONF_API_KEY: import_config[CONF_API_KEY],
+                CONF_MAIN_USER: None,
+                CONF_USERS: import_config[CONF_USERS],
+            },
+        )
+
+
+class LastFmOptionsFlowHandler(OptionsFlowWithConfigEntry):
+    """LastFm Options flow handler."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize form."""
+        errors = {}
+        if user_input is not None:
+            valid_users = []
+            for username in user_input[CONF_USERS]:
+                lastfm_user = get_lastfm_user(
+                    self._config_entry.data[CONF_API_KEY], username
+                )
+                lastfm_errors = validate_lastfm_user(lastfm_user)
+                if lastfm_errors:
+                    errors = lastfm_errors
+                else:
+                    valid_users.append(username)
+            user_input[CONF_USERS] = valid_users
+            if not errors:
+                return self.async_create_entry(
+                    title="LastFM",
+                    data={
+                        **self._config_entry.data,
```

Comment:
```suggestion
                        **self.options
```
You don't store in data, you store in options now